src/java.base/share/classes/sun/launcher/SecuritySettings.java

```diff
@@ -0,0 +1,213 @@
+
+    static void printSecuritySettings(String arg, PrintStream stream) {
+        ostream = stream;
+        switch (arg) {
+            case "properties" -> printSecurityProperties();
+            case "providers"  -> printSecurityProviderConfig(true);
+            case "tls"        -> printSecurityTLSConfig(true);
+            case "all"        -> printAllSecurityConfig();
+            default           -> ostream.println(
+                    "\nUnrecognized security subcommand. Valid values are " +
+                    "\"all\", \"properties\", \"providers\", \"tls\". See \"java -X\"\n");
+        }
+    }
+
+    // A non-verbose description of some core security configuration settings
+    static void printSecuritySummarySettings(PrintStream stream) {
+        ostream = stream;
+        ostream.println("Security settings summary: " + "\n" +
+                INDENT + "See \"java -X\" for verbose security settings options");
+        printSecurityProviderConfig(false);
+        printSecurityTLSConfig(false);
+    }
+
+    static void printAllSecurityConfig() {
+        ostream.println("Security settings:");
+        printSecurityProperties();
+        printSecurityProviderConfig(true);
+        printSecurityTLSConfig(true);
+    }
+
+    private static void printSecurityProperties() {
+        ostream.println(INDENT + "Security properties:");
+        Properties p = SharedSecrets.getJavaSecurityPropertiesAccess().getInitialProperties();
+        for (String key : p.stringPropertyNames().stream().sorted().toList()) {
+            String val = p.getProperty(key);
+            if (val.length() > 60) {
+                splitLongPropertyLines(key, val);
+            } else {
+                ostream.println(TWOINDENT + key + "=" + val);
+            }
+        }
+        ostream.println();
+    }
+
+    private static void splitLongPropertyLines(String key, String val) {
+        // split long property values which use well known separator
+        if (val.contains(",") || val.contains(";")) {
+            String separator = (val.contains(",")) ? "," : ";";
+            ostream.println(TWOINDENT + key + "=");
+            String[] values = val.split(separator);
+            String lastValue = values[values.length -1].trim();
+            List.of(values).forEach(
+                    s -> ostream.println(THREEINDENT + s.trim() +
+                            (s.trim().equals(lastValue) ? "" : separator)));
+        } else {
+            ostream.println(TWOINDENT + key + "=" + val);
+        }
+    }
+
+    private static void printSecurityTLSConfig(boolean verbose) {
+        SSLSocket ssls;
+        try {
+            ssls = (SSLSocket)
+                    SSLContext.getDefault().getSocketFactory().createSocket();
+        } catch (IOException | NoSuchAlgorithmException e) {
+            throw new InternalError("Failed to create SSL socket");
+        }
+
+        ostream.println(INDENT + "Security TLS configuration:");
```

Comment:
What about also noting the name of the TLS/JSSE provider in this line, for example:

"Security TLS configuration (SunJSSE provider):"

This would be useful information if a customer is using a 3rd party JSSE provider, and it is selected before SunJSSE, as the defaults in that case may be different.

Reply:
good suggestion Sean. Patch updated.

Reply:
I think we need to harden this area of code actually. Some JDK configurations may not have an SSL provider.

e.g. (removing JSSE from config)   

```
    See "java -X" for verbose security settings options
    Security provider static configuration: (in order of preference)
        Provider name: SUN
        Provider name: SunRsaSign
Exception in thread "main" java.lang.InternalError: Failed to create SSL socket
	at java.base/sun.launcher.SecuritySettings.printSecurityTLSConfig(SecuritySettings.java:121)
	at java.base/sun.launcher.SecuritySettings.printSecuritySummarySettings(SecuritySettings.java:75)
	at java.base/sun.launcher.LauncherHelper.showSettings(LauncherHelper.java:188)

```